feat(discover2): Enable browser right-click functions on SavedQuery cards

leedongwei opened the pull request:
## Before
![image](https://user-images.githubusercontent.com/1748388/72835686-09e81680-3c40-11ea-8349-53a9171a0a9f.png)

## After
![image](https://user-images.githubusercontent.com/1748388/72835812-4287f000-3c40-11ea-970c-0232277ef2fe.png)

leedongwei committed:
feat(discover2): Enable browser right-click functions on SavedQuery cards

markstory commented on src/sentry/static/sentry/app/views/eventsV2/querycard.tsx:
```diff
@@ -56,7 +61,7 @@ class QueryCard extends React.PureComponent<Props> {
 }
 
-const StyledQueryCard = styled('button')`
+const StyledQueryCard = styled(Link)`
```
Will this trigger DOM nesting warnings when the context menu is opened?

leedongwei replied:
Nope, the parent elements are `div` all the way up. I've checked in Chrome and FF.

dashed reviewed (approved):
Looks good to me 👍

leedongwei commented:
Rebased against `origin/master` and force-pushed because Percy was lighting up on every page. 

I had modified `link.jsx` which is used all over Sentry and I wanted to be sure it wasn't me.
